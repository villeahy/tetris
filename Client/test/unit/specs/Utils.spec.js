import {
  blockGenerator,
  clearLine,
  dropLines,
  checkLines
} from '@/components/tetris/gameUtils'

test('block generator test', () => {
  expect(blockGenerator(0)).toMatchSnapshot()
  expect(blockGenerator(1)).toMatchSnapshot()
  expect(blockGenerator(2)).toMatchSnapshot()
  expect(blockGenerator(3)).toMatchSnapshot()
  expect(blockGenerator(4)).toMatchSnapshot()
  expect(blockGenerator(5)).toMatchSnapshot()
  expect(blockGenerator(6)).toMatchSnapshot()
  expect(blockGenerator('asd')).toMatchSnapshot()
})

test('testing clear line function', () => {
  const gameBoard = [[0, 0, 0, 6], [0, 0, 5, 2], [0, 0, 0, 8]]
  expect(clearLine({ i: 3, gameBoard })).toMatchSnapshot()
})

test('test drop lines function', () => {
  const gameBoard = [[0, 0, 2, 0], [0, 2, 2, 0], [2, 2, 2, 0]]
  expect(dropLines({ i: 2, gameBoard })).toMatchSnapshot()
})

test('test of checklines function', () => {
  const gameBoard = [[2, 2, 2, 0], [0, 2, 2, 0], [0, 0, 2, 0]]
  expect(checkLines({ gameBoard })).toMatchSnapshot()
})
