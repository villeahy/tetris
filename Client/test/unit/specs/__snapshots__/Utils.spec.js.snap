// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block generator test 1`] = `
Array [
  Object {
    "i": 0,
    "y": 4,
  },
  Object {
    "i": -1,
    "y": 4,
  },
  Object {
    "i": 0,
    "y": 5,
  },
  Object {
    "i": -1,
    "y": 5,
  },
]
`;

exports[`block generator test 2`] = `
Array [
  Object {
    "i": -2,
    "y": 4,
  },
  Object {
    "i": -1,
    "y": 4,
  },
  Object {
    "i": 0,
    "y": 4,
  },
  Object {
    "i": 0,
    "y": 5,
  },
]
`;

exports[`block generator test 3`] = `
Array [
  Object {
    "i": -2,
    "y": 5,
  },
  Object {
    "i": -1,
    "y": 5,
  },
  Object {
    "i": 0,
    "y": 4,
  },
  Object {
    "i": 0,
    "y": 5,
  },
]
`;

exports[`block generator test 4`] = `
Array [
  Object {
    "i": -1,
    "y": 4,
  },
  Object {
    "i": -1,
    "y": 5,
  },
  Object {
    "i": 0,
    "y": 5,
  },
  Object {
    "i": 0,
    "y": 6,
  },
]
`;

exports[`block generator test 5`] = `
Array [
  Object {
    "i": -1,
    "y": 5,
  },
  Object {
    "i": -1,
    "y": 6,
  },
  Object {
    "i": 0,
    "y": 4,
  },
  Object {
    "i": 0,
    "y": 5,
  },
]
`;

exports[`block generator test 6`] = `
Array [
  Object {
    "i": -3,
    "y": 4,
  },
  Object {
    "i": -2,
    "y": 4,
  },
  Object {
    "i": -1,
    "y": 4,
  },
  Object {
    "i": 0,
    "y": 4,
  },
]
`;

exports[`block generator test 7`] = `
Array [
  Object {
    "i": -2,
    "y": 4,
  },
  Object {
    "i": -1,
    "y": 4,
  },
  Object {
    "i": -1,
    "y": 5,
  },
  Object {
    "i": 0,
    "y": 4,
  },
]
`;

exports[`block generator test 8`] = `
Array [
  Object {
    "i": 0,
    "y": 4,
  },
  Object {
    "i": -1,
    "y": 4,
  },
  Object {
    "i": 0,
    "y": 5,
  },
  Object {
    "i": -1,
    "y": 5,
  },
]
`;

exports[`test drop lines function 1`] = `
Array [
  Array [
    0,
    0,
    0,
    2,
  ],
  Array [
    0,
    0,
    2,
    2,
  ],
  Array [
    0,
    2,
    2,
    2,
  ],
]
`;

exports[`test of checklines function 1`] = `
Array [
  Array [
    0,
    2,
    2,
    0,
  ],
  Array [
    0,
    0,
    2,
    0,
  ],
  Array [
    0,
    0,
    0,
    0,
  ],
]
`;

exports[`testing clear line function 1`] = `
Array [
  Array [
    0,
    0,
    0,
    0,
  ],
  Array [
    0,
    0,
    5,
    0,
  ],
  Array [
    0,
    0,
    0,
    0,
  ],
]
`;
